pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1622,6 +1631,27 @@ func (s *AWSCloud) listSubnetIDsinVPC(vpcId string) ([]string, error) {
 	return subnetIds, nil
 }
 
+func isSubnetPublic(rt []*ec2.RouteTable, subnetID string) bool {
+	for _, table := range rt {
+		var found bool
+		for _, assoc := range table.Associations {
+			if orEmpty(assoc.SubnetId) == subnetID {
```

Comment:
We should probably move from `orEmpty` to the more standard (and equivalent) `aws.StringValue` function.  My fault (I don't know if it wasn't there at first or whether I just didn't find it) but let's try to have new code use aws.StringValue.  No need to change this though.